DummyServer/Controllers/BuildController.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BuildNotifications.Plugin.DummyServer;
+using BuildNotifications.PluginInterfaces.Builds;
+using DummyServer.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DummyServer.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class BuildController : ControllerBase
+    {
+        private readonly IDataStorage _dataStorage;
+
+        public BuildController(IDataStorage dataStorage)
+        {
+            _dataStorage = dataStorage;
+        }
+
+        [HttpGet]
+        public IEnumerable<WebBuild> Get()
+        {
+            return _dataStorage.Builds().Select(b => new WebBuild(b));
+        }
+
+        [HttpGet]
+        [Route(nameof(Extern))]
+        public IEnumerable<Build> Extern()
+        {
+            return _dataStorage.Builds();
+        }
+
+        [HttpPost]
+        public void Post([FromBody] WebBuild webBuild)
+        {
+            var existingBuild = _dataStorage.Builds().FirstOrDefault(b => b.Id.Equals(webBuild.Id));
+            if (existingBuild == null)
+                _dataStorage.AddBuild(webBuild.BranchName, webBuild.DefinitionName, webBuild.UserName);
+            else
+            {
+                existingBuild.Status = webBuild.Status;
+                existingBuild.Reason = webBuild.Reason;
+                existingBuild.LastChangedTime = DateTime.Now;
+            }
+        }
+
+        [HttpDelete]
+        public void Delete(string id)
+        {
+            _dataStorage.DeleteBuild(id);
+        }
+
+        [HttpGet]
+        [Route(nameof(Permutate))]
+        public void Permutate() => _dataStorage.PermutateBuilds();
```

Comment:
I would suggest a POST for this since it changes data on the server.
A GET should, well only get things